include/qthread/qthread.hpp

```diff
@@ -14,7 +14,11 @@
 template <bool> class OnlyTrue;
 template <> class OnlyTrue<true>
 {};
+#if __cplusplus >= 201103
```

Comment:
I suspect the case you encountered was a little different, because the behavior is the same in C.  A preprocessor that got that wrong would break a lot of code out there, and wouldn't last long before they had to fix that preprocessor.